script/signin.js

```diff
@@ -0,0 +1,124 @@
+/* 이메일 input에서 focus out 할 때
+값이 없을 경우 아래에 "이메일을 입력해 주세요." 에러 메시지를 보입니다 */
+
+const emailInput = document.getElementById('email');
+const emailErrorMessage = document.createElement('div');
+emailErrorMessage.className = 'error-message'; //에러 메시지를 표현하는 폼은 모두 같다.
+emailErrorMessage.classList.add('text-medium');
+emailErrorMessage.innerText = '이메일을 입력해 주세요'
+
+emailInput.addEventListener('focusout', function(){
+  if(!emailInput.value.trim()) { //trim의 값이 거짓일 때는 (빈 문자열 또는 공백만 포함일떄!) 즉, 지금 상태는 빈문자열일때!
+    emailErrorMessage.style.display = 'block'; // 화면에 보여줌
+    emailInput.classList.add('error-box');
+    emailInput.parentElement.lastElementChild.after(emailErrorMessage);
+  } else {
+    emailErrorMessage.style.display = 'none'; // 화면에 안보여줌
+    emailInput.classList.remove('error-box');
+  }
+});
+
+/*이메일 input에서 focus out 할 때, 이메일 형식에 맞지 않는 값이 있는 경우 아래에 
+"올바른 이메일 주소가 아닙니다" 라고 에러 메세지를 보냅니다. */
+
+const emailFormatErrorMessage = document.createElement('div');
+emailFormatErrorMessage.className = 'error-message';
+emailFormatErrorMessage.classList.add('text-medium');
+emailFormatErrorMessage.innerText = '올바른 이메일 주소가 아닙니다.';
+
+emailInput.addEventListener('focusout', function () {
+    const emailValue = emailInput.value.trim(); 
+    if (emailValue && !isValidEmail(emailValue)) { //빈문자열이 아니면서 이메일 형식과 맞지 않을 때 
+        emailFormatErrorMessage.style.display = 'block'; //화면에 보여줌
+        emailInput.parentElement.lastElementChild.after(emailFormatErrorMessage);
+    } else {
+        emailFormatErrorMessage.style.display = 'none';
+    }

```

Comment:
if-else 문을 불가피하게 사용해야하는 경우를 제외하고는 사용하지 않는 것이 가독성 측면에서 더 좋습니다. else 문 대신 [early return](https://devscb.tistory.com/130)을 사용해보시면 좋습니다.

```suggestion
    if (emailValue && !isValidEmail(emailValue)) { //빈문자열이 아니면서 이메일 형식과 맞지 않을 때 
        emailFormatErrorMessage.style.display = 'block'; //화면에 보여줌
        emailInput.parentElement.lastElementChild.after(emailFormatErrorMessage);
        return;
    } 
      emailFormatErrorMessage.style.display = 'none';

```
얼리 리턴은 유명한 리팩토링 기법인데요, 
if문을 사용하여 indent(코드 들여쓰기)가 깊어지면 가독성 측면에서 좋지 않기도 하고, 미세하게 코드 성능도 안좋기도 합니다. 그래서 이럴 때 빠른 return 문을 사용해서 코드를 개선해 볼 수 있어요.

Reply:
Commit ID : 2bb8a31f693ff5c4bd7865e25371ef21e9d660ad
정정했습니다! :D